framework/src/main/java/org/checkerframework/framework/util/CheckerMain.java

```diff
@@ -432,6 +450,29 @@ protected void addMainToArgs(final List<String> args) {
             args.add(quote(concatenatePaths(ppOpts)));
         }
 
+        if (PluginUtil.getJreVersion() > 8) {
+            // Get Patch_Module_argfile
+            final File patchModuleArgsFile = new File(jdkJar, "Patch_Modules_argfile");
```

Comment:
Can you describe the format of this file somewhere?

Reply:
Please see the comment below. Thanks.

Reply:
In the example file that you sent it looks like you patched every single module. I think it would make sense to limit that to only the modules that actually contain useful information. That is, if you didn't annotate a particular module, don't patch that module in - it will not have an effect and we don't need to distribute that module.

Reply:
One approach is to provide every module, even those that don't contain any changes.
The list of provided modules would only change when the Checker Framework supports a new JDK.

Another approach is to determine which modules contain annotations (maybe by grepping for `import org.checkerframework`?), and only provide those modules.
The list of patched modules could change for any Checker Framework release -- whenever the Checker Framework developers annotate more of the JDK.

The former seems simpler, and requires less-frequent work by Checker Framework users.
What are the key benefits of the latter that motivate it?  (Efficiency, or the size of the distribution, or something else?)

Reply:
Not sure whether this is for Shubham or me and nearly missed it...

As file `Patch_Modules_argfile` is generated, there shouldn't be an impact on users - they either use `CheckerMain` or import that argfile.

I assume we don't have annotations for many of the JDK modules, so distributing all of them seems wasteful and misleading. Maybe that assumption is wrong. I would think most of the current annotations ended up in `java.base.jmod` and some in management and xml.
@the1derer Can you already determine in which modules the current annotations ended up in?

Including only the ones that are `@AnnotatedFor` for one of the type systems would make the most sense to me, but if most modules are (partially) annotated I'm fine with simply listing all.

Reply:
@wmdietl Thanks for the clarification. Wasteful in terms of what?  Run time, bandwidth, or something else?  What is the impact for users?

@wmdietl @the1derer How is `Patch_Modules_argfile` generated?  My main concern was extra steps (and the ability to forget them!) when a CF developer adds new JDK annotations.  I was willing to make the distribution larger in order to avoid those steps and mistakes.  But I don't have concrete numbers to base that decision on.